core/src/main/java/com/radixdlt/rev2/REv2StateComputer.java

```diff
@@ -154,13 +154,9 @@ public void addToMempool(MempoolAdd mempoolAdd, NodeId origin) {
             transaction -> {
               try {
                 stateComputer.getMempoolInserter().addTransaction(transaction);
-
-                // TODO: Implement this event in the RustMempool. This requires a JNI -> Java
-                // callback
-                // TODO: interface which hasn't been implemented yet.
                 var success =
                     MempoolAddSuccess.create(
-                        RawNotarizedTransaction.create(transaction.getPayload()), null, origin);
+                        RawNotarizedTransaction.create(transaction.getPayload()), origin);
```

Comment:
Hmmm so:
* If we add new transactions to the mempool from other nodes (successfully), we create a `MempoolAddSuccess` here?
* But if it's via the Core API, we create a `MempoolAddSuccess` from the `mempoolRelayDispatcher`

This feels a little fragile that we originate this from two different places, but that's probably fine for now!

Separately, do you think it's also worth us adding a comment here that we only send MempoolAdd messages when we add the transaction to our mempools? (ie it's valid and not a duplicate) - to prevent infinite flood fill.